### ISSUE TITLE
``httpport`` attribute can not work well in ubuntu diskless 

### ISSUE THREAD

**Issue by hu-weihua** [xcat2/xcat-core#5862]
``httpport`` attribute can not work well in ubuntu diskless 

```
RUN:netstat -nlp|grep  -E "apache2|httpd" [Mon Dec  3 03:56:49 2018]
ElapsedTime:0 sec
RETURN rc = 0
OUTPUT:
tcp6       0      0 :::80                   :::*                    LISTEN      25584/apache2
CHECK:output =~ 80      [Pass]

RUN:lsdef -t site  -i httpport -c |awk -F"=" '{print $2}' [Mon Dec  3 03:56:49 2018]
ElapsedTime:0 sec
RETURN rc = 0
OUTPUT:


RUN:/opt/xcat/share/xcat/tools/autotest/testcase/httpport/change_http_port.sh 80 8898 [Mon Dec  3 03:56:49 2018]
ElapsedTime:2 sec
RETURN rc = 0
OUTPUT:
The original httpd port is 80 in /etc/apache2/ports.conf
start to change httpd listen port to 8898

RUN:netstat -nlp|grep  -E "apache2|httpd" [Mon Dec  3 03:56:51 2018]
ElapsedTime:0 sec
RETURN rc = 0
OUTPUT:
tcp6       0      0 :::8898                 :::*                    LISTEN      1124/apache2
CHECK:output =~ 8898    [Pass]
CHECK:rc == 0   [Pass]

RUN:chdef -t site httpport=8898 [Mon Dec  3 03:56:51 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
1 object definitions have been created or modified.
CHECK:rc == 0   [Pass]

RUN:lsdef -t site  -i httpport -c [Mon Dec  3 03:56:52 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
clustersite: httpport=8898
CHECK:output =~ 8898    [Pass]
CHECK:rc == 0   [Pass]


RUN:genimage  ubuntu18.04.1-ppc64el-netboot-compute [Mon Dec  3 03:58:15 2018]
ElapsedTime:156 sec
RETURN rc = 1
OUTPUT:
Generating image:
cd /opt/xcat/share/xcat/netboot/ubuntu; ./genimage -a ppc64el -o ubuntu18.04.1 -p compute --srcdir "/install/ubuntu18.04.1/ppc64el" --pkglist /opt/xcat/share/xcat/netboot/ubuntu/compute.ubuntu18.04.ppc64el.pkglist --otherpkgdir "/install/post/otherpkgs/ubuntu18.04.1/ppc64el" --postinstall "/opt/xcat/share/xcat/netboot/ubuntu/compute.postinstall" --rootimgdir /install/netboot/ubuntu18.04.1/ppc64el/compute --tempfile /tmp/xcat_genimage.1500 ubuntu18.04.1-ppc64el-netboot-compute
Run cmd [debootstrap --verbose --arch ppc64el bionic /install/netboot/ubuntu18.04.1/ppc64el/compute/rootimg http://ports.ubuntu.com/ubuntu-ports/] to create rootimage bootstraps
I: Retrieving InRelease
I: Checking Release signature
I: Valid Release signature (key id 790BC7277767219C42C86F933B4FE6ACC0B21F32)
I: Retrieving Packages
I: Validating Packages
I: Resolving dependencies of required packages...
I: Resolving dependencies of base packages...
I: Found additional base dependencies: bc
.....
The specified kerneldir /install/kernels does not exist!
The specified otherpkgdir /install/post/otherpkgs/ubuntu18.04.1/ppc64el does not exist!
Err:1 http://10.3.9.3/install/ubuntu18.04.1/ppc64el bionic InRelease
  Could not connect to 10.3.9.3:80 (10.3.9.3). - connect (111: Connection refused)
Reading package lists...
W: Failed to fetch http://10.3.9.3/install/ubuntu18.04.1/ppc64el/dists/bionic/InRelease  Could not connect to 10.3.9.3:80 (10.3.9.3). - connect (111: Connection refused)
W: Some index files failed to download. They have been ignored, or old ones used instead.
Reading package lists...
Building dependency tree...
Calculating upgrade...
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
.......
E: Unable to locate package dnsutils
E: Unable to locate package chrony
Failed to install packages bash ifupdown nfs-common openssl isc-dhcp-client libc-bin openssh-server openssh-client wget vim rsync busybox-static gawk dnsutils tar gzip xz-utils cpio chrony.

CHECK:rc == 0   [Failed]


```

**Comment by immarvin** [comment]
fixed in  https://github.com/xcat2/xcat-core/pull/5865

**Comment by hu-weihua** [comment]
@immarvin,  After  http port change to 8898, the  diskless installation still can not work in ubuntu (both 16.4.x and 18.4.x. ``genimage`` pass, but the installation failed. Can get message from console like below. It seems 80 still works in provision progress. 

```
root@f6u13k16:/opt/xcat/share/xcat/tools/autotest/result# tabdump site|grep 8898
"httpport","8898",,

root@f6u13k16:/opt/xcat/share/xcat/tools/autotest/result# netstat -nat |grep 8898
tcp6       0      0 :::8898                 :::*                    LISTEN

root@f6u13k16:# rinstall f6u13k18 osimage=ubuntu18.04.1-ppc64el-netboot-compute
```
The output from console:

```
.....
--2018-12-05 02:22:28--  http://f6u13k16//install/netboot/ubuntu18.04.1/ppc64el/compute/rootimg.cpio.gz
Resolving f6u13k16 (f6u13k16)... 10.6.13.16
Connecting to f6u13k16 (f6u13k16)|10.6.13.16|:80... failed: Connection refused.
Getting http://f6u13k16://install/netboot/ubuntu18.04.1/ppc64el/compute/rootimg.cpio.gz...
--2018-12-05 02:22:31--  http://f6u13k16//install/netboot/ubuntu18.04.1/ppc64el/compute/rootimg.cpio.gz
Resolving f6u13k16 (f6u13k16)... 10.6.13.16
Connecting to f6u13k16 (f6u13k16)|10.6.13.16|:80... failed: Connection refused.
^CGetting http://f6u13k16://install/netboot/ubuntu18.04.1/ppc64el/compute/rootimg.cpio.gz...
--2018-12-05 02:22:33--  http://f6u13k16//install/netboot/ubuntu18.04.1/ppc64el/compute/rootimg.cpio.gz
......
```

**Comment by hu-weihua** [comment]
Can not fix during 2.14.5. move to 2.14.6

**Comment by hu-weihua** [comment]
Have been verified against xcat 20190322.0615 daily build. close this issue.
